application/common/models/Reset.php

```diff
@@ -188,16 +186,59 @@ public function sendMethod()
 
     public function sendEmail()
     {
-        /**
-         * @todo send email to $this->method->value with reset code
+        /*
+         * send email to $this->method->value with reset code
          */
     }
 
+    /**
+     * Send phone verification and store resulting code
+     * @throws \Exception
+     */
     public function sendPhone()
     {
-        /**
-         * @todo use phone service component to send notification to $this->method->value
-         */
+        // Initialize log
+        $log = [
+            'action' => 'reset',
+            'method' => 'phone',
+            'method_id' => $this->method_id,
+            'previous_attempts' => $this->attempts,
+        ];
+
+        // Get phone number without comma
+        $number = $this->method->getRawPhoneNumber();
+
+        // Generate random code for potential use
+        $code = Utils::getRandomDigits(\Yii::$app->phone->codeLength);
+
+        // Call component send() method to send verification and capture resulting code
+        $result = \Yii::$app->phone->send($number, $code);
+
+        // Update db with code and increased attempts count
+        $this->code = $result;
+        $this->attempts++;
```

Comment:
Would it be worth updating the attempts count before we send the code to their phone number? As it currently stands, it looks like someone could attempt an unlimited number of times if there was some problem that was preventing the Reset from being updated in the database.